logviewer/src/main/java/org/togetherjava/logwatcher/DatabaseProvider.java

```diff
@@ -0,0 +1,70 @@
+package org.togetherjava.logwatcher;
+
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.FatalBeanException;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.togetherjava.tjbot.db.Database;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+@Configuration
+public class DatabaseProvider {
+
+    private static final AtomicReference<Database> ATOM_DB = new AtomicReference<>();
+    private static final Lock latch = new ReentrantLock();
```

Comment:
Why are those two variables static? Can't you just always create an instance in the constructor?
(And potentially mark the DatabaseProvider as a singleton so Spring is happy?)

Reply:
I only looked at `@Singleton` Annotation and found nothing that would belong to Spring.

It's now marked with the appropriate Scope